middle_end/flambda/simplify/simplify_set_of_closures.rec.ml

```diff
@@ -884,7 +891,9 @@ let simplify_lifted_sets_of_closures dacc ~orig_bound_symbols ~orig_static_const
     List.map
       (fun ({ code_ids = _; closure_symbols; }
              : Bound_symbols.Code_and_set_of_closures.t) ->
-        Closure_id.Map.map Name_in_binding_pos.symbol closure_symbols)
+        Closure_id.Lmap.map Name_in_binding_pos.symbol closure_symbols
+        |> Closure_id.Lmap.bindings
+        |> Closure_id.Map.of_list)
```

Comment:
Calling `map` then `of_list` seems inefficient. Would it make sense to use `fold` instead of `map` and build the `Closure_id.Map.t` directly during folding ?

Reply:
I find the map+of_list version a lot cleaner, personally.